static/app/components/newAssigneeSelectorDropdown.tsx

```diff
@@ -0,0 +1,448 @@
+import {Fragment} from 'react';
+import styled from '@emotion/styled';
+import uniqBy from 'lodash/uniqBy';
+
+import {assignToActor, clearAssignment} from 'sentry/actionCreators/group';
+import {openInviteMembersModal} from 'sentry/actionCreators/modal';
+import {AssigneeAvatar} from 'sentry/components/assigneeSelector';
+import type {SuggestedAssignee} from 'sentry/components/assigneeSelectorDropdown';
+import {Button} from 'sentry/components/button';
+import {Chevron} from 'sentry/components/chevron';
+import {
+  CompactSelect,
+  type SelectOption,
+  type SelectOptionOrSection,
+} from 'sentry/components/compactSelect';
+import IdBadge from 'sentry/components/idBadge';
+import LoadingIndicator from 'sentry/components/loadingIndicator';
+import {IconAdd} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import ConfigStore from 'sentry/stores/configStore';
+import MemberListStore from 'sentry/stores/memberListStore';
+import ProjectsStore from 'sentry/stores/projectsStore';
+import {useLegacyStore} from 'sentry/stores/useLegacyStore';
+import {space} from 'sentry/styles/space';
+import type {Actor, Group, SuggestedOwnerReason, Team, User} from 'sentry/types';
+import {buildTeamId} from 'sentry/utils';
+import {useMutation} from 'sentry/utils/queryClient';
+import type RequestError from 'sentry/utils/requestError/requestError';
+import useOrganization from 'sentry/utils/useOrganization';
+
+const suggestedReasonTable: Record<SuggestedOwnerReason, string> = {
+  suspectCommit: t('Suspect Commit'),
+  ownershipRule: t('Ownership Rule'),
+  projectOwnership: t('Ownership Rule'),
+  // TODO: codeowners may no longer exist
+  codeowners: t('Codeowners'),
+};
+
+export type OnAssignCallback = (
+  type: Actor['type'],
+  assignee: User | AssignableTeam,
+  suggestedAssignee?: SuggestedAssignee
+) => Promise<void>;
+
+type AssignableTeam = {
+  display: string;
+  email: string;
+  id: string;
+  team: Team;
+};
+
+type AssignableEntity = {
+  assignee: User | AssignableTeam;
+  id: string;
+  type: Actor['type'];
+};
+
+export interface NewAssigneeSelectorDropdownProps {
+  group: Group;
+  memberList?: User[];
+  noDropdown?: boolean;
+  onAssign?: OnAssignCallback;
+  onClear?: () => void;
+  owners?: Omit<SuggestedAssignee, 'assignee'>[];
+}
+
+function NewAssigneeSelectorDropdown({
+  group,
+  memberList,
+  noDropdown = false,
+  onAssign,
+  onClear,
+  owners,
+}: NewAssigneeSelectorDropdownProps) {
+  const organization = useOrganization();
+  const memberLists = useLegacyStore(MemberListStore);
+  const sessionUser = ConfigStore.get('user');
+
+  const {mutate, isLoading} = useMutation<
```

Comment:
isLoading doesn't seem to be working properly - when you click on a new assignee, the loading state doesn't appear, and it instead just pauses for a sec before rendering the new assignee. Did I do something wrong here? `isFetching` wouldn't pass the type checker but I found another use case that used isLoading.